src/bytes.rs

```diff
@@ -120,6 +120,31 @@ impl Default for CharIterator {
     }
 }
 
+impl Iterator for CharIterator {
+    type Item = char;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.eof() || self.position.offset >= self.length {
+            return None;
+        }
+
```

Comment:
Can you maybe add a comment here, saying that this is safe to do because `self.buffer` has the same length as `self.u8_buffer` and it is checked above?

I always think this is blowing up, when we are using `[]` to index on an vec/array